### PR TITLE
Fix/baseline repo

### PR DESCRIPTION
**Pull request by guillep**
Baselines have a strange method that accesses the monticello working copies trying to guess the repository they were loaded from.

This PR replaces this method to use the metacello registry instead.
This way, we remove one dependency from metacello->monticello.

### CODE REVIEW

**Comment by guillep** [comment]
Moreover, we did on purpose to give priority to `tonel://`repositories, so we can now remove the methods `packageRepository` from its subclasses: (`BaselineOfSpec2` and `BaselineOfWelcomeBrowser`) because they do the same.

**Comment by jecisc** [comment]
```
#repositoryDescriptions was sent to nil

UndefinedObject(Object)>>doesNotUnderstand: #repositoryDescriptions

BaselineOfPharo(BaselineOf)>>packageRepository

BaselineOfPharo(BaselineOf)>>packageRepositoryURL

BaselineOfPharo>>baseline:

[ self configuration perform: pragma methodSelector with: self ] in MetacelloBaselineConstructor(MetacelloAbstractVersionConstructor)>>evaluatePragma: in Block: [ self configuration perform: pragma methodSel[..]

FullBlockClosure(BlockClosure)>>ensure:

MetacelloBaselineConstructor(MetacelloAbstractVersionConstructor)>>evaluatePragma:

MetacelloBaselineConstructor>>calculate:project:

MetacelloBaselineConstructor(MetacelloAbstractVersionConstructor)>>on:project:

MetacelloBaselineConstructor class(MetacelloAbstractVersionConstructor class)>>on:project:

BaselineOfPharo(ConfigurationOf)>>project

BaselineOfPharo class(ConfigurationOf class)>>project

MetacelloScriptEngine>>getBaselineProjectUnconditionalLoad:

```

**Comment by guillep** [comment]
Greenish :') 


imagen

File: src/Metacello-Base/BaselineOf.class.st [hunk 1]
```diff
@@ -147,12 +147,23 @@ BaselineOf >> baseline: spec [
...
+	| newProject |
+	newProject := super newProjectWithRepositoryDescription:
+		           aListOrRepositoryDescriptions.
+	aListOrRepositoryDescriptions do: [ :desc |
+		newProject version spec repository: desc ].
+	^ newProject
+]
+
 { #category : 'accessing' }
 BaselineOf >> packageRepository [
```
Comment (jecisc) [inline src/Metacello-Base/BaselineOf.class.st:162]: Maybe the old way could be deprecated in a future PR to indicate the new way?

**Review by jecisc — APPROVED** [review]

File: src/Metacello-Base/BaselineOf.class.st (hunk 1 above)
Comment (guillep) [reply src/Metacello-Base/BaselineOf.class.st:162]: Yes yes it's in the plan :)

**Pull request merged**